crates/router/src/connector/checkout/transformers.rs

```diff
@@ -230,13 +251,57 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for PaymentsRequest {
                 Ok(a)
             }
             api::PaymentMethodData::Wallet(wallet_data) => match wallet_data {
-                api_models::payments::WalletData::GooglePay(_)
-                | api_models::payments::WalletData::ApplePay(_) => {
+                api_models::payments::WalletData::GooglePay(_) => {
                     Ok(PaymentSource::Wallets(WalletSource {
                         source_type: CheckoutSourceTypes::Token,
-                        token: item.get_payment_method_token()?,
+                        token: match item.get_payment_method_token()? {
+                            types::PaymentMethodToken::Token(token) => token,
+                            types::PaymentMethodToken::ApplePayDecrypt(_) => {
+                                Err(errors::ConnectorError::InvalidWalletToken)?
+                            }
+                        },
                     }))
                 }
+                api_models::payments::WalletData::ApplePay(_) => {
+                    let payment_method_token = item
+                        .payment_method_token
+                        .to_owned()
+                        .get_required_value("payment_token")
+                        .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+                    let decrypt_data = match payment_method_token {
+                        types::PaymentMethodToken::Token(_) => {
+                            Err(errors::ConnectorError::InvalidWalletToken)?
+                        }
+                        types::PaymentMethodToken::ApplePayDecrypt(data) => data,
+                    };
+                    let expiry_year_4_digit = format!(
+                        "20{}",
+                        decrypt_data.application_expiration_date.peek()[0..2].to_owned()
```

Comment:
Avoid accessing strings/slices by index, this can cause panics. Prefer to use `get()` instead.

Also, make the same changes for other similar usages here.